### DISCUSSION TITLE
Renaming a class deletes its history, is this a feature or a bug?

### DISCUSSION THREAD

**Discussion by istinnstudio** [apache/netbeans#7564] [General]
When a class is being renamed and then refactored, NetBeans deletes its history, is this a feature or a bug? Someone will have to remember the original name in order to get history back!

**Comment by matthiasblaesing** [comment]
Rightclick on parent folder -> History -> Revert Deleted

![image](https://github.com/user-attachments/assets/ac4f9205-d6f8-4614-8081-ba7d8d3eba8a)

**Reply by istinnstudio** [reply]
that is pretty strange, and not obvious, so rename is actually a new class copy of original and the original is being deleted. I wonder what is the technical aspect of such a choice. There should be a warning at least in refactor dialog.

**Reply by matthiasblaesing** [reply]
It is the most obvious way to implement a history function. You intercept the filesystem calls and make a copy on change. From the filesystem view you make a copy of the file, modify it and delete the original file.

No there should not be a warning in the refactor dialog. It does exactly what it says.

Don't rely on local history, you a real VCS (git being the most obvious one)! Local history is a tool to help you fix local screw ups, not a full blown VCS.

**Reply by istinnstudio** [reply]
clear, thanks
